src/data/polynomial/derivative.lean

```diff
@@ -185,18 +185,11 @@ end comm_semiring
 
 section domain
 variables [integral_domain R]
--- TODO: golf this, dunno how i broke it so bad
 lemma mem_support_derivative [char_zero R] (p : polynomial R) (n : ℕ) :
   n ∈ (derivative p).support ↔ n + 1 ∈ p.support :=
-begin
-rw finsupp.mem_support_iff, split; intro h,
-suffices h1 : p.coeff (n+1) ≠ 0, simp; tauto, contrapose! h,
-convert coeff_derivative _ _, simp [h],
-contrapose! h, simp,
-suffices : p.to_fun (n + 1) * (n + 1) = 0, simp only [mul_eq_zero] at this, cases this,
-{ exact this }, { norm_cast at this },
-erw ← h, symmetry, convert coeff_derivative _ _,
-end
+suffices (¬(coeff p (n + 1) = 0 ∨ ((n + 1:ℕ) : R) = 0)) ↔ coeff p (n + 1) ≠ 0,
+  by simpa only [mem_support_iff_coeff_ne_zero, coeff_derivative, ne.def, mul_eq_zero],
+by rw [nat.cast_eq_zero]; simp only [nat.succ_ne_zero, or_false]
```

Comment:
```suggestion
by { rw [nat.cast_eq_zero], simp only [nat.succ_ne_zero, or_false] }
```
We generally reserve `;` for when there are actually multiple goals being acted on.